### ISSUE TITLE
TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary

### ISSUE THREAD

**Issue by aRI0U** [ray-project/ray#27160]
### What happened + What you expected to happen

I just installed ray tune in a Python 3.10 conda environment using `pip install 'ray[tune]'` and I tried to run the PyTorch example from [this tutorial](https://docs.ray.io/en/latest/tune/examples/tune-pytorch-cifar.html#tune-pytorch-cifar-ref), but I had the following error:
```
$ python main.py
2022-07-28 11:30:50,957 ERROR services.py:1488 -- Failed to start the dashboard: Failed to start the dashboard, return code 1
 The last 10 lines of /tmp/ray/session_2022-07-28_11-30-49_098027_34758/logs/dashboard.log:
    return Server(migration_thread_pool, () if handlers is None else handlers,
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/site-packages/grpc/aio/_server.py", line 53, in __init__
    self._server = cygrpc.AioServer(
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 838, in grpc._cython.cygrpc.AioServer.__init__
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/asyncio/locks.py", line 77, in __init__
    super().__init__(loop=loop)
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/asyncio/mixins.py", line 17, in __init__
    raise TypeError(
TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary
2022-07-28 11:30:50,957 ERROR services.py:1489 -- Failed to start the dashboard, return code 1
 The last 10 lines of /tmp/ray/session_2022-07-28_11-30-49_098027_34758/logs/dashboard.log:
    return Server(migration_thread_pool, () if handlers is None else handlers,
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/site-packages/grpc/aio/_server.py", line 53, in __init__
    self._server = cygrpc.AioServer(
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 838, in grpc._cython.cygrpc.AioServer.__init__
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/asyncio/locks.py", line 77, in __init__
    super().__init__(loop=loop)
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/asyncio/mixins.py", line 17, in __init__
    raise TypeError(
TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary
Traceback (most recent call last):
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/site-packages/ray/_private/services.py", line 1465, in start_dashboard
    raise Exception(err_msg + last_log_str)
Exception: Failed to start the dashboard, return code 1
 The last 10 lines of /tmp/ray/session_2022-07-28_11-30-49_098027_34758/logs/dashboard.log:
    return Server(migration_thread_pool, () if handlers is None else handlers,
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/site-packages/grpc/aio/_server.py", line 53, in __init__
    self._server = cygrpc.AioServer(
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 838, in grpc._cython.cygrpc.AioServer.__init__
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/asyncio/locks.py", line 77, in __init__
    super().__init__(loop=loop)
  File "/home/alain/miniconda3/envs/ts/lib/python3.10/asyncio/mixins.py", line 17, in __init__
    raise TypeError(
TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary
[2022-07-28 11:31:02,057 E 34758 34758] core_worker.cc:137: Failed to register worker 01000000ffffffffffffffffffffffffffffffffffffffffffffffff to Raylet. IOError: [RayletClient] Unable to register worker with raylet. No such file or directory
```
I tried to uninstall/reinstall and so on but it does not work at all. Do you have an idea where this issue could come from?

### Versions / Dependencies

- Ray: 1.13.0 (installed with `pip`)
- PyTorch: 1.11.0 (installed with `conda`)
- Python: 3.10.2
- OS: Ubuntu 18.04

### Reproduction script

The script is the one from [this tutorial](https://docs.ray.io/en/latest/tune/examples/tune-pytorch-cifar.html#tune-pytorch-cifar-ref)

```python
import numpy as np
import os
import torch
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim
from filelock import FileLock
from torch.utils.data import random_split
import torchvision
import torchvision.transforms as transforms
import ray
from ray import tune
from ray.tune.schedulers import ASHAScheduler


def load_data(data_dir="./data"):
    transform = transforms.Compose([
        transforms.ToTensor(),
        transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
    ])

    # We add FileLock here because multiple workers will want to
    # download data, and this may cause overwrites since
    # DataLoader is not threadsafe.
    with FileLock(os.path.expanduser("~/.data.lock")):
        trainset = torchvision.datasets.CIFAR10(
            root=data_dir, train=True, download=True, transform=transform)

        testset = torchvision.datasets.CIFAR10(
            root=data_dir, train=False, download=True, transform=transform)

    return trainset, testset


class Net(nn.Module):
    def __init__(self, l1=120, l2=84):
        super(Net, self).__init__()
        self.conv1 = nn.Conv2d(3, 6, 5)
        self.pool = nn.MaxPool2d(2, 2)
        self.conv2 = nn.Conv2d(6, 16, 5)
        self.fc1 = nn.Linear(16 * 5 * 5, l1)
        self.fc2 = nn.Linear(l1, l2)
        self.fc3 = nn.Linear(l2, 10)

    def forward(self, x):
        x = self.pool(F.relu(self.conv1(x)))
        x = self.pool(F.relu(self.conv2(x)))
        x = x.view(-1, 16 * 5 * 5)
        x = F.relu(self.fc1(x))
        x = F.relu(self.fc2(x))
        x = self.fc3(x)
        return x


def train_cifar(config, checkpoint_dir=None):
    net = Net(config["l1"], config["l2"])

    device = "cpu"
    if torch.cuda.is_available():
        device = "cuda:0"
        if torch.cuda.device_count() > 1:
            net = nn.DataParallel(net)
    net.to(device)

    criterion = nn.CrossEntropyLoss()
    optimizer = optim.SGD(net.parameters(), lr=config["lr"], momentum=0.9)

    # The `checkpoint_dir` parameter gets passed by Ray Tune when a checkpoint
    # should be restored.
    if checkpoint_dir:
        checkpoint = os.path.join(checkpoint_dir, "checkpoint")
        model_state, optimizer_state = torch.load(checkpoint)
        net.load_state_dict(model_state)
        optimizer.load_state_dict(optimizer_state)

    data_dir = os.path.abspath("./data")
    trainset, testset = load_data(data_dir)

    test_abs = int(len(trainset) * 0.8)
    train_subset, val_subset = random_split(
        trainset, [test_abs, len(trainset) - test_abs])

    trainloader = torch.utils.data.DataLoader(
        train_subset,
        batch_size=int(config["batch_size"]),
        shuffle=True,
        num_workers=8)
    valloader = torch.utils.data.DataLoader(
        val_subset,
        batch_size=int(config["batch_size"]),
        shuffle=True,
        num_workers=8)

    for epoch in range(10):  # loop over the dataset multiple times
        running_loss = 0.0
        epoch_steps = 0
        for i, data in enumerate(trainloader, 0):
            # get the inputs; data is a list of [inputs, labels]
            inputs, labels = data
            inputs, labels = inputs.to(device), labels.to(device)

            # zero the parameter gradients
            optimizer.zero_grad()

            # forward + backward + optimize
            outputs = net(inputs)
            loss = criterion(outputs, labels)
            loss.backward()
            optimizer.step()

            # print statistics
            running_loss += loss.item()
            epoch_steps += 1
            if i % 2000 == 1999:  # print every 2000 mini-batches
                print("[%d, %5d] loss: %.3f" % (epoch + 1, i + 1,
                                                running_loss / epoch_steps))
                running_loss = 0.0

        # Validation loss
        val_loss = 0.0
        val_steps = 0
        total = 0
        correct = 0
        for i, data in enumerate(valloader, 0):
            with torch.no_grad():
                inputs, labels = data
                inputs, labels = inputs.to(device), labels.to(device)

                outputs = net(inputs)
                _, predicted = torch.max(outputs.data, 1)
                total += labels.size(0)
                correct += (predicted == labels).sum().item()

                loss = criterion(outputs, labels)
                val_loss += loss.cpu().numpy()
                val_steps += 1

        # Here we save a checkpoint. It is automatically registered with
        # Ray Tune and will potentially be passed as the `checkpoint_dir`
        # parameter in future iterations.
        with tune.checkpoint_dir(step=epoch) as checkpoint_dir:
            path = os.path.join(checkpoint_dir, "checkpoint")
            torch.save(
                (net.state_dict(), optimizer.state_dict()), path)

        tune.report(loss=(val_loss / val_steps), accuracy=correct / total)
    print("Finished Training")


def test_best_model(best_trial):
    best_trained_model = Net(best_trial.config["l1"], best_trial.config["l2"])
    device = "cuda:0" if torch.cuda.is_available() else "cpu"
    best_trained_model.to(device)

    checkpoint_path = os.path.join(best_trial.checkpoint.value, "checkpoint")

    model_state, optimizer_state = torch.load(checkpoint_path)
    best_trained_model.load_state_dict(model_state)

    trainset, testset = load_data()

    testloader = torch.utils.data.DataLoader(
        testset, batch_size=4, shuffle=False, num_workers=2)

    correct = 0
    total = 0
    with torch.no_grad():
        for data in testloader:
            images, labels = data
            images, labels = images.to(device), labels.to(device)
            outputs = best_trained_model(images)
            _, predicted = torch.max(outputs.data, 1)
            total += labels.size(0)
            correct += (predicted == labels).sum().item()


    print("Best trial test set accuracy: {}".format(correct / total))


def main(num_samples=10, max_num_epochs=10, gpus_per_trial=2):
    config = {
        "l1": tune.sample_from(lambda _: 2 ** np.random.randint(2, 9)),
        "l2": tune.sample_from(lambda _: 2 ** np.random.randint(2, 9)),
        "lr": tune.loguniform(1e-4, 1e-1),
        "batch_size": tune.choice([2, 4, 8, 16])
    }
    scheduler = ASHAScheduler(
        max_t=max_num_epochs,
        grace_period=1,
        reduction_factor=2)
    result = tune.run(
        tune.with_parameters(train_cifar),
        resources_per_trial={"cpu": 2, "gpu": gpus_per_trial},
        config=config,
        metric="loss",
        mode="min",
        num_samples=num_samples,
        scheduler=scheduler
    )

    best_trial = result.get_best_trial("loss", "min", "last")
    print("Best trial config: {}".format(best_trial.config))
    print("Best trial final validation loss: {}".format(
        best_trial.last_result["loss"]))
    print("Best trial final validation accuracy: {}".format(
        best_trial.last_result["accuracy"]))

    if ray.util.client.ray.is_connected():
        # If using Ray Client, we want to make sure checkpoint access
        # happens on the server. So we wrap `test_best_model` in a Ray task.
        # We have to make sure it gets executed on the same node that
        # ``tune.run`` is called on.
        from ray.util.ml_utils.node import force_on_current_node
        remote_fn = force_on_current_node(ray.remote(test_best_model))
        ray.get(remote_fn.remote(best_trial))
    else:
        test_best_model(best_trial)


if __name__ == "__main__":
    main(num_samples=2, max_num_epochs=2, gpus_per_trial=0)
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by krfricke** [comment]
cc @simon-mo looks like this could be dashboard-related?

**Comment by simon-mo** [comment]
@alanwguo

**Comment by alanwguo** [comment]
We don't use the loop parameter in any of the asyncio Lock's in the ray codebase. This is likely a downstream dependency issue. Might be related to https://github.com/grpc/grpc/issues/27888

@aRI0U , what version of grpcio do you have installed?

**Comment by alanwguo** [comment]
Yes, I confirmed a install of `ray[tune]` in a fresh conda environment will install the right version of grpcio and the tutorial code runs fine.

When I installed grpcio==1.41.1, I got the same error as you.

I'll put up a PR to pin grpcio >= 1.42.0 for py310.

**Comment by aRI0U** [comment]
Hi,
Thanks a lot for your reply, in fact I had `grpcio==1.35.0`. When I upgrade it works like a charm.
I also tried with more recent versions of `grpcio`:
- With versions `1.42.0` to `1.45.0`, it perfectly works
- With versions `1.46` and `1.47.0`, it works but at the end of the tuning I have this mysterious message:
```
E0801 17:18:11.681294530   16630 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
E0801 17:18:11.724995363   16630 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
```
- With version `1.48.0` (latest), the program stops just before the call to `tune.run()`, and even trying to interrupt it with `ctrl+C` does not work (I had to `ctrl+Z` then manually kill the process).

These experiments have been conducted only on the tutorial so results have to be taken with caution.

Thanks again for your help!
